src/plugins/stake-plugins/thorchainSavers/tcSaversPlugin.ts

```diff
@@ -644,6 +701,40 @@ const headers = {
   'Content-Type': 'application/json'
 }
 
+// ----------------------------------------------------------------------------
+// Estimate the fees to unstake by faking an unstake calculation using the
+// address and amount from the largest staked address in the savers pool for
+// the requested asset. This will calculate the withdrawBps for the fake
+// unstake by using the ratio of the requested unstake amount compared to the
+// total amount of the largest staked address
+// ----------------------------------------------------------------------------
+const estimateUnstakeFee = async (opts: EdgeGuiPluginOptions, request: ChangeQuoteRequest, asset: string, ninerealmsClientId: string): Promise<string> => {
+  const { currencyCode, stakePolicyId, wallet } = request
+  const saversResponse = await fetchWaterfall(thornodeServers, `thorchain/pool/${asset}/savers`, { headers: { 'x-client-id': ninerealmsClientId } })
+  if (!saversResponse.ok) {
+    const responseText = await saversResponse.text()
+    throw new Error(`Thorchain could not fetch /pool/savers: ${responseText}`)
+  }
+  const saversJson = await saversResponse.json()
+  const savers = asSavers(saversJson)
+
+  const bigSaver = savers.reduce((prev, current) => (gt(current.units, prev.units) ? current : prev))
```

Comment:
if `savers.length === 0` then this will throw.

Reply:
👍